buildSrc/subprojects/ide/src/main/kotlin/org/gradle/gradlebuild/ide/IdePlugin.kt

```diff
@@ -77,230 +56,120 @@ limitations under the License."""
 open class IdePlugin : Plugin<Project> {
 
     override fun apply(project: Project): Unit = project.run {
-        configureIdeaForRootProject()
+        if (providers.environmentVariable("idea.active").getOrElse("false").toBoolean()) {
+            configureIdeaForRootProject()
+        }
     }
 
     private
     fun Project.configureIdeaForRootProject() {
+        val rootProject = this
         apply(plugin = "org.jetbrains.gradle.plugin.idea-ext")
-        apply(plugin = "idea")
-        tasks.named("idea") {
-            doFirst {
-                throw RuntimeException("To import in IntelliJ, please follow the instructions here: https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md#intellij")
+        with(the<IdeaModel>()) {
+            module {
+                excludeDirs = setOf(intTestHomeDir)
             }
-        }
-
-        val rootProject = this
-        plugins.withType<IdeaPlugin> {
-            with(model) {
-                module {
-                    excludeDirs = excludeDirs + rootExcludeDirs
+            project {
+                settings {
+                    configureCopyright()
+                    configureRunConfigurations(rootProject)
+                    doNotDetectFrameworks("android", "web")
                 }
+            }
+        }
+    }
 
-                project {
-                    jdkName = "11.0"
```

Comment:
Shouldn't we still set all of this? The JDK name is import, as is the `Git` VCS. I don't know about the wildcards, i.e. if the `.gradle` directory is excluded by default.

Reply:
This is all information that can be derived from the project already and IDEA does that. I tested this locally and IDEA detects everything correctly. I assume it knows that we are in a git repo and reads the git config. Java version is read from the source/target configuration in the Gradle.

Reply:
Did you try this after deleting your `.idea` directory? If the `vcs.xml` file isn't there, it doesn't seem that IDEA infers the `Git` VCS. At least I needed to add it manually after I the `vcs.xml` file on a merge conflict.

I'll try out the JDK inference on a new checkout.

Reply:
Both things seem to work on a new checkout. 🤷

Reply:
Thanks for taking the time to confirm!